core/src/test/java/org/elasticsearch/common/unit/TimeValueTests.java

```diff
@@ -125,6 +128,30 @@ public void testRoundTrip() {
         assertThat(TimeValue.parseTimeValue(t.getStringRep(), null, "test"), equalTo(t));
     }
 
+    private static final String FRACTIONAL_TIME_VALUES_ARE_NOT_SUPPORTED = "fractional time values are not supported";
+
+    public void testNonFractionalTimeValues() {
+        final String s = randomAsciiOfLength(10) + randomTimeUnit();
+        final ElasticsearchParseException e =
+            expectThrows(ElasticsearchParseException.class, () -> TimeValue.parseTimeValue(s, null, "test"));
+        assertThat(e, hasToString(containsString("failed to parse [" + s + "]")));
+        assertThat(e, not(hasToString(containsString(FRACTIONAL_TIME_VALUES_ARE_NOT_SUPPORTED))));
+        assertThat(e.getCause(), instanceOf(NumberFormatException.class));
+    }
+
+    public void testFractionalTimeValues() {
+        final String s = Double.toString(randomDouble() / 10) + randomTimeUnit();
```

Comment:
I don't think you need to divide at all - randomDouble is between 0 and 1, I believe.